cmd/clusterctl/pkg/log/logger.go

```diff
@@ -0,0 +1,276 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+package log implements a clusterctl friendly logr.Logger derived from
+https://github.com/kubernetes/klog/blob/master/klogr/klogr.go.
+
+The logger is designed to work in two scenarios:
+
+1. clusterctl CLI, with logs printed to stdout with a formatting that is easy to read for users
+   but also simple to parse for identifying specific values.
+
+2. clusterctl library, with log entries sent to a channel for allowing different types of consumption.
```

Comment:
Just noticed that you dropped the channel support. This comment will need to change.